File: app/src/main/java/app/bolling/chucknorris/dagger/RoomModule.java
package app.bolling.chucknorris.dagger;

import android.app.Application;
import android.arch.persistence.room.Room;

import javax.inject.Singleton;

import app.bolling.chucknorris.database.AppDatabase;
import app.bolling.chucknorris.database.dao.JokeDao;
import dagger.Module;
import dagger.Provides;

@Module
public class RoomModule {

    @Singleton
    @Provides
    AppDatabase providesRoomDatabase(Application app) {
        return  Room.databaseBuilder(app, AppDatabase.class, "chuck-db").fallbackToDestructiveMigration().build();
    }

    @Singleton
    @Provides
    JokeDao providesProductDao(AppDatabase demoDatabase) {
        return demoDatabase.jokeDao();
    }
}

File: app/src/main/java/app/bolling/chucknorris/dagger/RoomModule.kt
package app.bolling.chucknorris.dagger

import android.app.Application
import android.arch.persistence.room.Room
import app.bolling.chucknorris.database.AppDatabase
import app.bolling.chucknorris.database.dao.JokeDao
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RoomModule {

    @Singleton
    @Provides
    internal fun providesRoomDatabase(app: Application): AppDatabase {
        return Room.databaseBuilder(app, AppDatabase::class.java, "chuck-db").fallbackToDestructiveMigration().build()
    }

    @Singleton
    @Provides
    internal fun providesProductDao(demoDatabase: AppDatabase): JokeDao {
        return demoDatabase.jokeDao()
    }
}